__tests__/main.test.js

```diff
@@ -0,0 +1,43 @@
+import {
+  createNewTodo,
+  createTodoControls,
+  todoArray
+} from '../main'
+
+const { JSDOM } = require("jsdom");
+const { window } = new JSDOM('<!DOCTYPE html><html><head></head><body></body></html>');
+const $ = require("jquery")(window);
+
+describe("add todo to list", () => {
+  beforeEach(() => {
+    $("body").html('<input type="text" value="" placeholder="What do you need ?" class="input-text" required></input>'
+      + '<button class="add-item">' +
+      '<i class="fa-regular fa-plus"></i>' +
+      '</button>'
+      + '<div class="todo"/>' + '<ul class="todo__list"></ul>'
+      + '</div>')
+  })
+  test('check data todo', () => {
+    expect(typeof createNewTodo()).toBe('object')
+  })
+
+  test('add Todo', () => {
+    $('.input-text').val("water")
+
+    const newTodo = createNewTodo($('.input-text').val())
```

Comment:
no, it shouldn't look like this. Let's check your function `createNewTodo`:
```
export const createNewTodo = text => {   
    const todo = $('<li>', {
        class: 'todo__item',
        text,
        'data-id': uuidv4()
    })

    return todo
}
```

as you can see in this function you're preparing todo to append it to DOM, so the test should look like following:
```
const text = 'Clean house'
const todo = createNewTodo(text)

expect(todo.classList.contains('todo__item')).toBe(true)
expect(todo.getAttribute('data-id')).toBeTruthy()
expect(todo.innerHTML).toEqual(text)
```

So basically, you're checking what your function was doing, you shouldn't write here additional code